### ISSUE TITLE
Access to message attributes

### ISSUE THREAD

**Issue by goofy2k** [knolleary/pubsubclient#861]
Hi,  is it possible to access more attributes (other than topic and payload) of a message?  I was thinking of the "retained" property. It may be helpful to treat an incoming message depending on this flag.  Also QoS, dup may be interesting.

**Comment by knolleary** [comment]
No, the retained flag is not exposed in the callback - only the topic and payload.

In our experience, the retained flag is rarely very useful.
